crates/rpc/rpc-eth-api/src/ext.rs

```diff
@@ -16,3 +19,20 @@ pub trait L2EthApiExt {
         condition: ConditionalOptions,
     ) -> RpcResult<B256>;
 }
+
+#[async_trait::async_trait]
+impl<T> L2EthApiExtServer for T
+where
+    T: FullEthApi,
+    jsonrpsee_types::error::ErrorObject<'static>: From<T::Error>,
+{
```

Comment:
we can't implement this default like this because we need to actually handle the conditionals.

can we start with a new helper type in 

https://github.com/paradigmxyz/reth/blob/5db01290f877ea4b8762c25a343bbcb6ea053ebb/crates/optimism/rpc/Cargo.toml#L2-L2

instead, that wraps `P: Pool`?

like `struct L2TransactionConditional<P>{ pool: P}`

Reply:
Based on the impl in op-geth, we don't check the conditional in the pool since that means repeated state checks as transactions are reinserted in the pool.

Instead we check the conditional at the rpc layer preemptively to the pool and at the block building process. The transaction has a mutable `rejected` [field](https://github.com/ethereum-optimism/op-geth/blob/optimism/core/types/transaction.go#L71), set by the that is used to drive eviction in the mempool (also being used in the interop feature)

still wrapping my head around the different modules integrate with each other in this code. If a similar mutable approach is possible with `OpTransactionSigned` or if there is a nicer idiomatic way for the payload builder to surface better evict txs from the mempool

Reply:
oh nice I didn't see #13806, I was making the same exact change to `OpPooledTransaction` to manage some external state. Let me pull that in.

I'll create a separate PR (hopefully tomorrow) so we can discuss from there separately from there. Can you briefly elaborate on why we'd need `L2TransactionConditional<P> { pool: P }`? I'm not fully following a change to the api. In the draft I have I simply have an `Option<TransactionConditional>`  field on `OpTransactionSigned` so afaik pool api should have to be wrapped at all since the pool transactions are already typed.

Unless you were thinking  `OpSignedTransaction` is untouched and the pool has a separate submission api for these txs.

I think a draft a small diff in the optimism crate will help clarify. Will get this out super soon!

Reply:
hey @mattsse, I opened this brief draft PR https://github.com/paradigmxyz/reth/pull/13926 with a few comments/questions.

Would appreciate your thoughts and I can dig into an implementation from there

Reply:
as @mattsse says, the conditionals need to be handled and they are specific of the L2, so it won't be implemented in this place of the code anyway. perhaps you would like to close this pr and give it another go by implementing https://github.com/paradigmxyz/reth/pull/13762/files#r1921021197  @oac1771 , unless @hamdiallam has already started with it?